apps/rush/src/actions/GenerateAction.ts

```diff
@@ -59,6 +67,22 @@ export default class GenerateAction extends BaseRushAction {
 
     const installManager: InstallManager = new InstallManager(this.rushConfiguration);
 
+    const shrinkwrapFile: ShrinkwrapFile | undefined
+        = ShrinkwrapFile.loadFromFile(this.rushConfiguration.committedShrinkwrapFilename);
+
+    if (shrinkwrapFile
+      && !this._forceParameter.value
+      && installManager.createTempModulesAndCheckShrinkwrap(shrinkwrapFile)) {
+      console.log();
+      console.log(colors.yellow('"rush generate" is unnecessary, since all' +
+        ' direct dependencies already exist in the shrinkwrap file.'));
+      console.log(colors.yellow('Did you mean to run "rush install"?'));
+      console.log();
+      console.log('You can use the \'--force\' flag to bypass this check and forcibly' +
+        ' generate a new shrinkwrap file.');
```

Comment:
I'm not sure how I like how we're telling them that they did something wrong.

Here's a possible alternative text:

Skipping generate, since all project dependencies are already satisfied. If you want to force an upgrade to the latest compatible versions, use “rush generate --force”.  Otherwise, just run “rush install”.